pwnlib/context/__init__.py

```diff
@@ -1251,7 +1261,8 @@ def cache_dir(self):
         if not os.access(cache, os.W_OK):
             return None
 
-        return cache
+        self._cache_dir = cache
```

Comment:
This is inconsistent with how everything inside of `context` works, we should not be setting any attributes that are not TLS-backed, since each thread may modify the context and they should not interfere with each other.

This is why there's a lot of work in the `context` module to work off of TLS.  Set the value you want in `pwnlib.context.context.defaults` if you really need a non-standard default .

Reply:
I've now taken the approach of setting the base path as an entry in `defaults` and generating the cache_dir as a property based on that value. `cache_base_dir` is a new validated property and changing its value clears the `cache_dir` stashed in `_tls`. The `cache_dir` is not a validated property since I didn't want to give it a setter without specific input from you in that direction. It just has a getter which reaches into `_tls` for a well known key (`"cache_dir"`, which should probably be a constant somewhere to avoid drift).

Let me know what you think of the new approach.